src/layers/conv.jl

```diff
@@ -925,3 +927,86 @@ function LSTMAggregator(args...; kwargs...)
     in_ch = args[1][1]
     return SAGEConv(args..., Flux.LSTMCell(in_ch, in_ch); kwargs...)
 end
+
+"""
+    EGNNConv((in_dim, int_dim, out_dim); init)
+"""
+
+struct EGNNConv <: MessagePassing
+   nn_edge
+   nn_x
+   nn_h
+
+   in_dim
+   int_dim
+   out_dim
+end
+
+@functor EGNNConv
+
+function EGNNConv(dims::NTuple{3,Int}; init=glorot_uniform)
+    in_dim, int_dim, out_dim = dims
+
+    m_len = in_dim * 2 + 2
+
+    nn_edge = Flux.Dense(m_len, int_dim; init=init)
+
+    nn_x = Flux.Dense(int_dim, 1; init=init)
+    nn_h = Flux.Dense(in_dim + int_dim, out_dim; init=init)
+
+    return EGNNConv(nn_edge, nn_x, nn_h, dims...)
+end
+
+function EGNNConv(nn_edge, nn_x, nn_h; init=glorot_uniform)
+
+    # Assume that these are strictly MLPs (no conv)
+    nn_edge.init(init)
+    nn_x.init(init)
+    nn_h.init(init)
+
+    in_dim = nn_edge.layers[1].W |> x->size(x)[2]
+    int_dim = nn_edge.layers[end].W |> x->size(x)[1]
+    out_dim = nn_h.layers[end].W |> x->size(x)[1]
+    return EGNNConv(nn_edge, nn_x, nn_h, in_dim, int_dim, out_dim)
+end
+
+function message(egnn::EGNNConv, v_i, v_j, e)
+    in_dim = egnn.in_dim
+    h_i = v_i[1:in_dim]
+    h_j = v_j[1:in_dim]
+
+    x_i = v_i[in_dim+1:end]
+    x_j = v_j[in_dim+1:end]
+
+    if isnothing(e)
+        a = 1
+    else
+        a = e[1]
+    end
+
+    input = vcat(h_i, h_j, norm(x_i - x_j, 2)^2, a)
+    edge_msg = egnn.nn_edge(input)
+    return vcat(edge_msg, (x_i - x_j) * egnn.nn_x(edge_msg)[1], 1)
+end
+
+function update(e::EGNNConv, m, h)
+    mi = m[1:e.int_dim]
+    x_msg = m[e.int_dim+1:end-1]
+    M = m[end]
+
+    C = 1/(M-1)
+
+    nn_node_out = e.nn_h(vcat(h[1:e.in_dim], mi))
+
+    z = zeros(e.out_dim + e.int_dim)
+    z[1:hout_size] = nn_node_out
+    z[hout_size+1:end] = h[e.in_dim+1:end] + C * x_msg
+    return z
+end
```

Comment:
`update` function should be only `nn_h` itself.

Reply:
Are you sure? The paper says that there are two outputs to EGCL - x_i^{l+1} and h_{i]^{l+1}.

Reply:
Sorry, I missed that. The paper mentions $h_i^{l+1}, x_i^{l+1} = EGCL(h^l, x^l, \mathcal{E})$, so we can have output $[h_i^{l+1}, x_i^{l+1}]$, i.e. `vcat(h_i, x_i)`.

Reply:
Right. If this code is correct then this should be what is happening currently